src/lib/systemd_journal/query.rb

```diff
@@ -40,70 +35,74 @@ class Query
     #   * An scalar value to be passed to the --boot argument of journalctl
     #   In the first two cases, the values can be Time objects or strings of
     #   any format accepted by journalctl for --until and --since
-    # @param filters [Hash] valid keys are :priority, :units and :matches, the
-    #   values are strings or array of strings with the format accepted by the
-    #   corresponding journalctl argument. If the value is an Array, the
+    # @param filters [Hash] The keys must match one of the VALID_FILTERS.
+    #   The values are strings or array of strings with the format accepted by
+    #   the corresponding journalctl argument. If the value is an Array, the
     #   argument will be repeated as many times as needed.
     def initialize(interval: nil, filters: {})
       @interval = interval
-      @filters = filters
+      # Make sure the keys are strings
+      @filters = Hash[filters.map {|k,v| [k.to_s, v] }]
```

Comment:
above you said that it must match one of VALID_FILTERS, but you do conversion before even if documentation explicitelly mention strings. Again it is over engineered and such hidden features make future refactoring much harder as some code can depend on it.